pkg/image/registry/imagestreammapping/rest.go

```diff
@@ -64,13 +65,27 @@ func (s *REST) Create(ctx kapi.Context, obj runtime.Object) (runtime.Object, err
 		tag = api.DefaultImageTag
 	}
 
-	if err := s.imageRegistry.CreateImage(ctx, &image); err != nil && !errors.IsAlreadyExists(err) {
-		return nil, err
+	imageCreateErr := s.imageRegistry.CreateImage(ctx, &image)
+	if imageCreateErr != nil && !errors.IsAlreadyExists(imageCreateErr) {
+		return nil, imageCreateErr
+	}
+
+	// prefer dockerImageReference set on image for the tagEvent if the image is new
+	ref := image.DockerImageReference
+	if errors.IsAlreadyExists(imageCreateErr) && image.Annotations[api.ManagedByOpenShiftAnnotation] == "true" {
+		// the image is managed by us and, most probably, tagged in some other image stream
+		// let's make the reference local to this stream
+		if streamRef, err := api.DockerImageReferenceForStream(stream); err == nil {
+			streamRef.ID = image.Name
+			ref = streamRef.Exact()
+		} else {
+			utilruntime.HandleError(err)
```

Comment:
I'm not really convinced one is more preferred than the other. In strategy and rest files under `pkg/image/registry/` the use of `utilruntime.HandleError()` vs `glog.V(?).Info` is 7 to 8 .

But I'll rewrite it.